### PR TITLE
Test server Nexus endpoint operator apis

### PR DESCRIPTION
**Pull request by pdoerner**
## What was changed

Updated API to `v1.36.0`
Added test server implementations of Nexus endpoint CRUD APIs to TestOperatorService

## Why?

To support implementing Nexus operations in the test server.

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: temporal-test-server/src/main/java/io/temporal/internal/testservice/TestNexusEndpointStore.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2024 Temporal Technologies, Inc. All Rights Reserved.
```
Comment (pdoerner) [inline temporal-test-server/src/main/java/io/temporal/internal/testservice/TestNexusEndpointStore.java:2]: Not sure why CI is complaining about the license. All I did was copy paste, remove Amazon and Uber, and update the date. But if I was supposed to leave those alone, lmk.

File: temporal-test-server/src/main/java/io/temporal/internal/testservice/TestNexusEndpointStoreImpl.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+  @Override
+  public Endpoint updateEndpoint(String id, long version, EndpointSpec spec) {
+    validateEndpointSpec(spec);
+
+    Endpoint prev = endpoints.get(id);
+
+    if (prev == null) {
+      throw Status.NOT_FOUND
+          .withDescription("Could not find Nexus endpoint with ID: " + spec.getName())
```
Comment (bergundy) [inline temporal-test-server/src/main/java/io/temporal/internal/testservice/TestNexusEndpointStoreImpl.java:64]:
```suggestion
          .withDescription("Could not find Nexus endpoint with ID: " + id)
```

File: temporal-test-server/src/main/java/io/temporal/internal/testservice/TestNexusEndpointStoreImpl.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    if (!ENDPOINT_NAME_REGEX.matcher(spec.getName()).matches()) {
+      throw Status.INVALID_ARGUMENT
+          .withDescription(
+              "Nexus endpoint name ("
+                  + spec.getName()
+                  + ") does not match expected pattern: "
+                  + ENDPOINT_NAME_REGEX.pattern())
+          .asRuntimeException();
+    }
+    if (!spec.hasTarget()) {
```
Comment (bergundy) [inline temporal-test-server/src/main/java/io/temporal/internal/testservice/TestNexusEndpointStoreImpl.java:151]: Let's only accept worker targets for now.

**Review by bergundy — COMMENTED** [review]
Overall LGTM. Will defer to @Quinn-With-Two-Ns for the approval.

File: temporal-test-server/src/main/java/io/temporal/internal/testservice/TestNexusEndpointStore.java (hunk 1 above)
Comment (Quinn-With-Two-Ns) [reply temporal-test-server/src/main/java/io/temporal/internal/testservice/TestNexusEndpointStore.java:2]: You just need to run ` ./gradlew licenseFormat`

**Comment by Quinn-With-Two-Ns** [comment]
Overall LGTM, are we able to add tests as well? Ideally we can add test that both verify the API behaves as expected, and behaves like the real server. Here is what I did for update https://github.com/temporalio/sdk-java/commit/6b39e44738f62611e8ddcb62a7f61e5996d355f4.

File: temporal-test-server/src/test/java/io/temporal/testserver/functional/NexusEndpointTest.java [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+import java.util.UUID;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class NexusEndpointTest {
+  @Rule public SDKTestWorkflowRule testWorkflowRule = SDKTestWorkflowRule.newBuilder().build();
+
+  @Before
+  public void checkExternal() {
```
Comment (pdoerner) [inline temporal-test-server/src/test/java/io/temporal/testserver/functional/NexusEndpointTest.java:48]: Opened https://github.com/temporalio/sdk-java/issues/2165 to remove skip once 1.25.0 is officially released

**Review by Quinn-With-Two-Ns — APPROVED** [review]

**Pull request merged**